### PR TITLE
Collection Library

### PR DESCRIPTION
**Pull request by CarsonF**
This is the follow up to #6569 

- `bolt/collection` library has been required
- Replace usages of `Arr` class with version from library
- Moved `Arr::deepDiff`
- Deprecated `Arr` class

I want to wait a few days to see how things follow in core before tagging a stable release on bolt/collection. We'll need to tag it before next beta though.

### CODE REVIEW

File: src/Logger/Handler/RecordChangeHandler.php [hunk 1]
```diff
@@ -211,4 +210,40 @@ private function initialize()
...
         $this->initialized = true;
     }
+
+    /**
+     * @param array $a
+     * @param array $b
+     *
+     * @return array [key, left, right][]
+     */
+    private function diff(array $a, array $b)
```
Comment (GawainLynch) [inline src/Logger/Handler/RecordChangeHandler.php]: Erm, you might have forgotten about B.C on [`Bolt\DeepDiff`](https://github.com/bolt/bolt/blob/release/3.3/src/DeepDiff.php#L25)

Comment (CarsonF) [inline src/Logger/Handler/RecordChangeHandler.php]: Errrumm what's a BC? 🤦‍♂️

Comment (CarsonF) [inline src/Logger/Handler/RecordChangeHandler.php]: Fixed using your suggestion of reflection

**Comment by bobdenotter** [comment]
Cool stuff! 🎊 🎉

**Pull request merged**